functions/public/Update-omsAgent.ps1

```diff
@@ -0,0 +1,179 @@
+		[System.Management.Automation.Credential()]
+		$Credential
+	)
+
+	Begin
+	{
+		$commonSessionParams = @{
+			ErrorAction = 'Stop'
+		}
+
+		If ($PSBoundParameters['Credential'])
+		{
+			$commonSessionParams.Credential = $Credential
+		}
+	}
+	Process
+	{
+		forEach ($computer in $computerName)
+		{
+			try
+			{
+
+				Write-Verbose "[$(Get-Date -Format G)] - $computer - Creating Remote PS Session"
+				$psSession = New-PSSession -ComputerName $computer -EnableNetworkAccess @commonSessionParams
+
+				Write-Verbose "[$(Get-Date -Format G)] - $computer - Checking if OMS is Installed"
+
+				$orginalAgent = Get-omsAgentInternal -computerName $computer -session $psSession
+				if(($orginalAgent))
+				{
+					If ($Pscmdlet.ShouldProcess($computer, 'Update OMS Agent'))
+					{
+						 $path = Invoke-Command -Session $pssession -ScriptBlock {
+							$path = Join-Path $ENV:temp "MMASetup.exe"
+
+							# Check if file exists and if so remove
+							if(Test-Path $path)
+							{
+								Remove-Item $path -force -Confirm:$false
+							}
+
+							$path
+						 }
+
+						if($PSBoundParameters.sourcePath) # Check for source path
+						{
+							Write-Verbose "[$(Get-Date -Format G)] - $computer - Copying files over powershell session"
+							Copy-Item -Path $sourcePath -Destination  $path -ToSession $psSession -Force
+						}
+						else
+						{
+							Write-Verbose "[$(Get-Date -Format G)] - $computer - Trying to download new installer from URL - $downloadURL"
+							Invoke-Command -Session $psSession -ScriptBlock {
+								Invoke-WebRequest $USING:downloadURL -OutFile $USING:path -ErrorAction Stop | Out-Null
+							} -ErrorAction Stop
+						}
+
+
+						Write-Verbose "$computer - Trying to Update OMS..."
+						$installString = $path + ' /Q:A /R:N /C:"setup.exe /qn AcceptEndUserLicenseAgreement=1"'
+
+						$installSuccess = Invoke-Command -Session $psSession -ScriptBlock {
+							Get-service HealthService|Stop-Service
+							cmd.exe /C $USING:installString
+							$LASTEXITCODE
+						} -ErrorAction Stop
+
+						Write-Verbose "[$(Get-Date -Format G)] - $computer - $installSuccess"
+
+						if($installSuccess -eq 3010)
+						{
+							Write-Verbose "$computer - OMS updated correctly based on the exit code"
+							Write-Verbose "$computer - Restarting HealthService"
+							Invoke-Command -Session $psSession -ScriptBlock {
+										Get-service HealthService|Start-Service
+												} -ErrorAction Stop
+
+						}
+						Elseif($installSuccess -ne 0)
+						{
+							Invoke-Command -Session $psSession -ScriptBlock {
+										Get-service HealthService|Start-Service
+												} -ErrorAction Stop
+							Write-Verbose "$computer - Restarting HealthService"
+							Write-Error "$computer - OMS didn't update correctly based on the exit code"
+
+
+						}
+						else
+						{
+							if((Get-omsAgentInternal -computerName $computer -session $psSession).displayverison -gt $orginalAgent.displayverison)
```

Comment:
Maybe add logic so if "(Get-omsAgentInternal -computerName $computer -session $psSession).displayverison -eq $orginalAgent.displayverison)" the service is started. Might make it a little more robust if something stopped the agent installing but didn't change the original install.